zerver/views/users.py

```diff
@@ -780,17 +784,28 @@ def get_member_backend(
 @typed_endpoint
 def get_members_backend(
     request: HttpRequest,
-    user_profile: UserProfile,
+    maybe_user_profile: UserProfile | AnonymousUser,
     *,
     user_ids: Json[list[int]] | None = None,
     include_custom_profile_fields: Json[bool] = False,
     client_gravatar: Json[bool] = True,
 ) -> HttpResponse:
+    if isinstance(maybe_user_profile, UserProfile):
+        user_profile = maybe_user_profile
+        realm = user_profile.realm
+    else:
+        realm = get_valid_realm_from_request(request)
+        if not realm.allow_web_public_streams_access():
+            raise MissingAuthenticationError
```

Comment:
Test failure is missing coverage here; fixing.

Reply:
Thanks